### PR TITLE
added carousel

### PR DESCRIPTION
**Pull request by silchyn**
[DEMO](https://silchyn.github.io/react_carousel/)

### CODE REVIEW

File: src/components/Carousel.tsx [hunk 1]
```diff
@@ -1,18 +1,114 @@
...
+          className="Carousel__list"
+          style={{
+            transform: `translateX(-${itemWidth * position}px)`,
+            width: itemWidth * images.length,
+            transition: `transform ${animationDuration}ms`,
+          }}
+        >
+          {
+            images
+              .map((image, i) => (
```
Comment (ivan-kadykalo) [inline src/components/Carousel.tsx:78]:
```suggestion
            images.map((image, i) => (
```

File: src/App.tsx [hunk 2]
```diff
@@ -20,17 +25,146 @@ class App extends React.Component<{}, State> {
...
+
+        target.value = this.state.step.toString();
+      } else if (target.classList.contains('form__animationDuration')) {
+        this.setState({ animationDuration: Math.max(+target.value, 0) });
+
+        target.value = this.state.animationDuration.toString();
+      } else if (target.classList.contains('form__infinite')) {
+        this.setState({ infinite: target.checked });
+      }
+    }
```
Comment (ivan-kadykalo) [inline src/App.tsx:75]: 🤯 Avoid nested ifs and use switch case

**Review by ivan-kadykalo — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (silchyn) [reply src/App.tsx:75]: > 🤯 Avoid nested ifs and use switch case

But how to use switch case with .contains()?

Comment (ivan-kadykalo) [reply src/App.tsx]: May google be with you🙃

image

File: src/components/Carousel.tsx [hunk 3]
```diff
@@ -1,18 +1,113 @@
...
+    const buttons = document.querySelector('.Carousel__buttons');
+
+    if (buttons) {
+      buttons.children[0].addEventListener('click', () => {
+        this.setState((state, props) => {
+          if (props.infinite && !state.position) {
+            return { position: props.images.length - props.frameSize };
+          }
+
+          return { position: Math.max(0, state.position - props.step) };
```
Comment (ivan-kadykalo) [inline src/components/Carousel.tsx:34]:
```suggestion
          return (props.infinite && !state.position) 
          	? { position: props.images.length - props.frameSize }
          	: { position: Math.max(0, state.position - props.step) };
```

**Review by ivan-kadykalo — CHANGES_REQUESTED** [review]
Fix switch case

File: src/App.tsx (hunk 2 above)
Comment (silchyn) [reply src/App.tsx]: > May google be with you🙃
> 
> image

Yeah, I know about that kludge but why it is better than if-else-if?

Comment (ivan-kadykalo) [reply src/App.tsx]: Switch case is more readable and works faster than if else.

Comment (ivan-kadykalo) [reply src/App.tsx]: It is not a better solution each time, but in this case it is.

**Review by Ihor-Karpyn — APPROVED** [review]

**Review by ivan-kadykalo — APPROVED** [review]

**Pull request closed without merging**